plugins/modules/backup_vault.py

```diff
@@ -0,0 +1,349 @@
+    """
+    try:
+        client.delete_backup_vault(BackupVaultName=vault_name)
+    except (BotoCoreError, ClientError) as err:
+        module.fail_json_aws(err, msg="Failed to delete the Backup Vault")
+
+
+def main():
+    argument_spec = dict(
+        state=dict(default="present", choices=["present", "absent", "enabled", "disabled"]),
+        backup_vault_name=dict(default="default", type="str"),
+        encryption_key_arn=dict(type="str"),
+        creator_request_id=dict(default="default", type="str"),
+        backup_vault_tags=dict(type="dict", aliases=["tags"]),
+        purge_tags=dict(default=True, type="bool"),
+    )
+
+    required_if = [("state", "present", ["backup_vault_name"]), ("state", "enabled", ["backup_vault_name"])]
+
+    module = AnsibleAWSModule(argument_spec=argument_spec, supports_check_mode=True, required_if=required_if)
+
+    # collect parameters
+    if module.params["state"] in ("present", "enabled"):
+        state = "present"
+    elif module.params["state"] in ("absent", "disabled"):
+        state = "absent"
+    tags = module.params["backup_vault_tags"]
+    purge_tags = module.params["purge_tags"]
+    ct_params = dict(
+        BackupVaultName=module.params["backup_vault_name"],
+        BackupVaultTags=module.params["backup_vault_tags"],
+        EncryptionKeyArn=module.params["encryption_key_arn"],
+        CreatorRequestId=module.params["creator_request_id"],
+    )
+
+    client = module.client("backup")
+    region = module.region
+
+    results = dict(changed=False, exists=False)
+
+    # Get existing backup vault facts
+    try:
+        vault = get_vault_facts(module, client, ct_params["BackupVaultName"])
+    except (BotoCoreError, ClientError) as err:
+        module.debug("Unable to get vault facts {0}".format(err))
+
+    # If the vault exists set the result exists variable
+    if vault is not None:
+        results["exists"] = True
+
+    if state == "absent" and results["exists"]:
+        # If Trail exists go ahead and delete
+        results["changed"] = True
+        results["exists"] = False
+        results["backupvault"] = dict()
+        if not module.check_mode:
+            delete_backup_vault(module, client, vault["BackupVaultName"])
+
+    elif state == "present" and not results["exists"]:
+        # Backup Vault doesn't exist just go create it
+        results["changed"] = True
+        results["exists"] = True
+        if not module.check_mode:
+            if tags:
+                ct_params["BackupVaultTags"] = tags
+            # If we aren't in check_mode then actually create it
+            create_backup_vault(module, client, ct_params)
+
+            # Get facts for newly created Backup Vault
+            vault = get_vault_facts(module, client, ct_params["BackupVaultName"])
+
+        # If we are in check mode create a fake return structure for the newly created vault
+        if module.check_mode:
+            acct_id = "123456789012"
```

Comment:
I personally do not like this approach. I would rather go by return and empty result or a message in check mode instead of a hardcoded value. @tremble what do you suggest?

Reply:
I agree with @alinabuzachis - really don't like hard coded "check_mode" information like this.